src/lib/internal/utils/occurrences.ts

```diff
@@ -0,0 +1,21 @@
+import { escapeRegExp } from "./escapeRegExp";
+
+const PATTERN_CACHE = new Map<string, RegExp>();
+
+/**
+ * Get the number of occurrences of a substring in a string.
+ * @param string The string.
+ * @param substring The substring.
+ * @returns The number of occurrences.
+ */
+export function occurrences(string: string, substring: string) {
+  let pattern: RegExp;
+  if (PATTERN_CACHE.has(substring)) {
+    // biome-ignore lint/style/noNonNullAssertion: <explanation>
+    pattern = PATTERN_CACHE.get(substring)!;
```

Comment:
Correct point, but from a macro perspective, it is better to use the non-null assertion operation here, since the newly created RegExp object must be cached.